src/lib.rs

```diff
@@ -10,6 +10,17 @@ struct LinscanIndex {
     index: index::Index
 }
 
+// convert milliseconds to a duration. If ms is infinity, then return None.
+fn ms_to_duration(ms_opt: Option<f32>) -> Option<Duration> {
```

Comment:
I still think we should make this a u32, then use Duration::from_millis() to convert to duration.

Reply:
but millis in `u32` would make it impossible to set a limit of less than one ms (which is useful for testing with smaller datasets). so switch to microseconds? besides, what is the benefit of using u32 instead of f32?

Reply:
I see - I didn't realize we may want to set a limit of less than 1ms. Thanks for explaining that!